packages/core/tests/__snapshots__/config.test.ts.snap

```diff
@@ -72,7 +72,7 @@ exports[`Should compose create Rsbuild config correctly > Merge Rsbuild config 1
           "pnpapi",
         ],
         "filename": {
-          "js": "[name].js",
+          "js": "[name].mjs",
```

Comment:
Why the change to mjs?

Reply:
Because of the directory of the test case, package.json was not resolved successfully here just like your issue before, so `autoExtension` was not enabled.

Reply:
Ah, looks like this will fix more bugs than just one ...
... do we have a unit test case for this?

Reply:
Current tests is enough since it correctly use the root package.json to generate extensions. This snapshot is used to check config merge behaviour, we do not find the issue of not finding package.json in this snapshot before. So this snapshot is just the unit test case for this.
 
And about autoExtension, there already have unit tests.